lms/resources/lti_launch.py

```diff
@@ -23,6 +23,34 @@ def __init__(self, request):
         self._request = request
         self._authority = self._request.registry.settings["h_authority"]
 
+    def get_or_create_course(self):
+        """Get the course this LTI launch based on the request's params."""
+        course_service = self._request.find_service(name="course")
+        params = self._request.parsed_params
+
+        tool_consumer_instance_guid = params["tool_consumer_instance_guid"]
+        context_id = params["context_id"]
+
+        # Generate the authority_provided_id based on the LTI
+        # tool_consumer_instance_guid and context_id parameters.
+        # These are "recommended" LTI parameters (according to the spec) that in
+        # practice are provided by all of the major LMS's.
+        # tool_consumer_instance_guid uniquely identifies an instance of an LMS,
+        # and context_id uniquely identifies a course within an LMS. Together they
+        # globally uniquely identify a course.
+        authority_provided_id = hashed_id(tool_consumer_instance_guid, context_id)
+
+        legacy_course = course_service.get_or_create(authority_provided_id)
+        course = course_service.upsert(
+            authority_provided_id,
+            context_id,
+            params["context_title"],
+            self._course_extra(),
+            legacy_course.settings,
+        )
+
+        return legacy_course, course
```

Comment:
Returns both, as we move away from legacy_course it will just deal with `couse` but I like how explicit this is and there's no many places where it's used ATM.